### PR TITLE
fix client.Do()

### PR DESCRIPTION
**Pull request by clearcodecn**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -122,9 +121,7 @@ func (c *client) Signature(reqSign *sign.RequestSignature) (string, error) {
 func (c *client) Do(ctx context.Context, method, url string, req ...interface{}) *Result {
 	// 1. serialize the request
 	var reqBuffer []byte
-	// now we only use method condition to solve this.
-	if len(req) > 0 && method != http.MethodGet &&
-		req[0] != nil && !reflect.ValueOf(req[0]).IsNil() {
```
Comment (gunsluo) [inline client.go:127]: please use `req[0] != nil && !reflect.ValueOf(req[0]).IsNil()` instread of `req[0] != nil`.    beasure of The request may be `(*PayRequest)(nil)`

File: refund_query.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func (r *RefundQueryRequest) Do(ctx context.Context, c Client) (*RefundQueryResp
 	}
 
 	resp := &RefundQueryResponse{}
-	if err := c.Do(ctx, http.MethodGet, url, r).Scan(resp); err != nil {
+	if err := c.Do(ctx, http.MethodGet, url, nil).Scan(resp); err != nil {
```
Comment (gunsluo) [inline refund_query.go:69]: if err := c.Do(ctx, http.MethodGet, url, nil).Scan(resp); err != nil { 
==> 
if err := c.Do(ctx, http.MethodGet, url).Scan(resp); err != nil {

File: client.go (hunk 1 above)
Comment (clearcodecn) [reply client.go:127]:
```
var req = (*PayRequest)(nil)
print(req == nil)   // true
```

**Pull request merged**